ios/Classes/SwiftFlutterWebAuthPlugin.swift

```diff
@@ -46,8 +46,18 @@ public class SwiftFlutterWebAuthPlugin: NSObject, FlutterPlugin {
                 let session = ASWebAuthenticationSession(url: url, callbackURLScheme: callbackURLScheme, completionHandler: completionHandler)
 
                 if #available(iOS 13, *) {
-                    guard let provider = UIApplication.shared.delegate?.window??.rootViewController as? FlutterViewController else {
-                        result(FlutterError(code: "FAILED", message: "Failed to aquire root FlutterViewController" , details: nil))
+                    if var topController = UIApplication.shared.keyWindow?.rootViewController {
+                        while let presentedViewController = topController.presentedViewController {
+                            topController = presentedViewController
+                        }
+                        if topController is UINavigationController {
+                            var viewControllerProvider = (topController as! UINavigationController).visibleViewController!
+                            session.presentationContextProvider = viewControllerProvider as! ASWebAuthenticationPresentationContextProviding
+                        } else {
+                            session.presentationContextProvider = topController as! ASWebAuthenticationPresentationContextProviding
+                        }
```

Comment:
I'm a bit weary about the force unwrapping here. Could we change it out for checked casts instead?

Reply:
Ok changed and pushed the commit.